nautobot/extras/jobs.py

```diff
@@ -283,6 +283,28 @@ def as_form(self, data=None, files=None, initial=None):
 
         return form
 
+    @classmethod
+    def deserialize_data(cls, data):
+        """
+        Given data input for a job execution, deserialize it by resolving object references using defined variables.
+
+        This converts a list of pk's back into a QuerySet for MultiObjectVar instances and single pk values into
+        model instances for ObjectVar.
+        """
+        vars = cls._get_vars()
+        return_data = {}
+
+        for field_name, value in data.items():
+            var = vars[field_name]
+            if isinstance(var, MultiObjectVar):
+                return_data[field_name] = var.field_attrs["queryset"].filter(pk__in=value)
```

Comment:
Should we fail in some fashion if the resulting queryset doesn't contain all requested PKs? (e.g. value is a list of 3 PKs but the queryset only contains 2 matching items because one of the given PKs doesn't exist)